### PR TITLE
Issue7

### PR DESCRIPTION
**Pull request by ekigamba**
1. Add descriptions
2. Add screenshots
3. Show flow
4. Update setup + other instructions
5. Customisations from Parent Library

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,27 +2,844 @@
...
-Tips and tricks (to be completed):
-=================================
+OpenSRP Client Native Form Module/app provides the ability to easily create Android forms using JSON forms. 
 
-* How to setup your Android SDK so that Maven finds it: http://pivotal.github.com/robolectric/maven-quick-start.html
+Thanks to this [Android Native JSON Form Library](https://github.com/vijayrawatsan/android-json-form-wizard) from which OpenSRP Client Native Form has been customised to fit OpenSRP Requirements
+
+> **NOTE**
+>
+> **JSON Form** is written using **JSON (syntax)** which can be found [here](https://www.w3schools.com/js/js_json_syntax.asp).
```
Comment (pld) [inline README.md]: let's not link to w3schools since that's a commercial site, maybe http://json.org/ is fine

File: README.md [hunk 2]
```diff
@@ -2,27 +2,844 @@
...
+
+# Features
+
+1. It enables definition of Android forms in JSON
+2. It enables one to define metadata for OpenMRS forms
+3. It enables one to define validations for form inputs in JSON
+   * [Regular Expression](https://en.wikipedia.org/wiki/Regular_expression) validation rules
+   * Number validation rules
+   * Alphabetic & alphanumeric validation rules in JSON
+4. It eanbles one to define form input constraints using JSON
```
Comment (pld) [inline README.md]: typo

File: README.md [hunk 3]
```diff
@@ -2,27 +2,844 @@
...
+   * Min value
+   * Max value
+5. It enables one to define OpenMRS mappings in JSON eg.
```
Comment (pld) [inline README.md]: remove trailing `eg` not sure why that's here

File: README.md [hunk 4]
```diff
@@ -2,27 +2,844 @@
...
+6. It enables one to define skip logic for fields based on values entered in other fields
+
+# App Walkthrough
+
+1. When you open the app, the following page is displayed
+
+![Main Page Screenshot](https://user-images.githubusercontent.com/31766075/30383189-ca377ca6-98a9-11e7-8c23-9538214a975f.png)
+![Main Page Screenshot -> Menu Open](https://user-images.githubusercontent.com/31766075/30383181-c49c8732-98a9-11e7-9f95-f56fc0cb7931.png)
+
+This page has a menu at the top-right which opens a sample Patient Registeration form written in JSON Form.
```
Comment (pld) [inline README.md]: typo

File: README.md [hunk 5]
```diff
@@ -2,27 +2,844 @@
...
+4. Validations & Constraints for input fields i.e.
+   * `v_regex` - This is used for validating input using [Regular Expression](https://en.wikipedia.org/wiki/Regular_expression) 
+   * `v_min` - This validation ensures input entered is not below minimum value stated
+   * `v_max` - Ensure that the input entered is not above the maximum value stated
+
+5. Different types of comparisons were added for the **Skip Logic**
+
+This enables a field to be shown only if the condition set in `'ex'` is `True`
+
+Skip Logic is defined using in `"relevance"` value
```
Comment (pld) [inline README.md]: "in" => "the"

File: README.md [hunk 6]
```diff
@@ -2,27 +2,844 @@
...
+```
+"relevance": {
+          "step1:Place_Birth": {
+            "type": "string",
+            "ex": "equalTo(., \"Health facility\")"
+          }
+        }
+```
+
+The field above will only be shown if the (key) `Place_Birth` value, which is **Patient's Place of Birth(Location)**, is equals to `"Health Facility"`
```
Comment (pld) [inline README.md]: remove parens around "key"

Comment (pld) [reply README.md]: * add "the" after "which is"
* add space after "Birth" before open paren

Comment (ekigamba) [reply README.md]: **Before:** The field above will only be shown if the (key) `Place_Birth` value, which is **Patient's Place of Birth(Location)**, is equals to `"Health Facility"`

**After:** The field above will only be shown if the key `Place_Birth` value, which is the **Patient's Place of Birth (Location)**, is equals to `"Health Facility"` 

After removing the parenthesis, the statement seems not right. Advise on this

Comment (pld) [reply README.md]: it does seem a little off, I'd rearrange as

> The field above will only be shown if the value for the key Place_Birth, which is the Patient's Place of Birth (Location), is equals to "Health Facility"

Comment (ekigamba) [reply README.md]: 👍

**Pull request merged**